File: Assets/Scripts/PuzzleGeneration/PuzzleLayout.cs
using System;
using System.Collections.Generic;
using UnityEngine.Serialization;

namespace PuzzleGeneration
{
    [Serializable]
    public class PuzzleLayout
    {
        public int rows;
        public int cols;
        public float width;
        public float height;
        public PieceShape shape;
        public List<PieceCut> initialPieceCuts;

        public PuzzleLayout(
            int rows, 
            int cols, 
            float width, 
            float height, 
            PieceShape shape, 
            List<PieceCut> initialPieceCuts
        )
        {
            this.rows = rows;
            this.cols = cols;
            this.width = width;
            this.height = height;
            this.shape = shape;
            this.initialPieceCuts = initialPieceCuts;
        }
    }
}

File: PuzzleServer/src/main/kotlin/com/vrpuzzle/puzzleserver/model/type/PuzzleLayout.kt
package com.vrpuzzle.puzzleserver.model.type

data class PuzzleLayout(
    val rows: Int,
    val cols: Int,
    val width: Float,
    val height: Float,
    val shape: PieceShape,
    val initialPieceCuts: List<PieceCut>,
)